### PR TITLE
content type change to text/plain + add tests for utils.FormatObject

### LINKED ISSUE #412: Change wrong context-type "application/text" to "text/plain"
1. We have a bunch of places using wrong context-type. Need to change them to correct values.
2. Add more test cases for utils.FormatObject. Output is not expected for some scenarios where format is yaml.

### PR DESCRIPTION
**Pull request by lirenjie95**
Fix #412

### CODE REVIEW

File: api/pkg/apis/v1alpha1/utils/utils_test.go [hunk 1]
```diff
@@ -69,26 +69,53 @@ func TestFormatObjectEmptyDict(t *testing.T) {
...
 	obj := map[string]interface{}{
 		"foo": "bar",
+		"man": "what  ",
+		"can": "\"I\"",
+		"say": "?",
 	}
 	val, err := FormatObject(obj, false, "$.foo", "")
 	assert.Nil(t, err)
 	assert.Equal(t, "\"bar\"", string(val))
+	val, err = FormatObject(obj, false, "$.man", "json")
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/utils/utils_test.go:83]: If I remember correctly, we have some issues in current object to YAML function. It cannot deal with YAML conversion in all cases. We need to check whether there's YAML go package and replace the logic in FormatObject.

Comment (lirenjie95) [reply api/pkg/apis/v1alpha1/utils/utils_test.go:83]: I replaced it with `gopkg.in/yaml.v2` in the new version

Comment (lirenjie95) [reply api/pkg/apis/v1alpha1/utils/utils_test.go:88]: I refactored the `FormatObject` function. The logic about "marshal obj into a json, and then unmashal back" is only for the input `obj` is an array. I moved it to the `isArray` condition branch.

File: api/pkg/apis/v1alpha1/utils/utils.go [hunk 2]
```diff
@@ -234,8 +227,10 @@ func FormatObject(obj interface{}, isArray bool, path string, format string) ([]
 				jData, _ = json.Marshal(ob)
 			}
 		}
+	} else {
+		jData, err = json.Marshal(obj)
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/utils/utils.go:231]: What if path == "" and format == "yaml"?

File: api/pkg/apis/v1alpha1/utils/utils.go [hunk 3]
```diff
@@ -234,8 +227,14 @@ func FormatObject(obj interface{}, isArray bool, path string, format string) ([]
 				jData, _ = json.Marshal(ob)
 			}
 		}
+	} else {
+		if format == "yaml" {
+			jData, err = yaml.Marshal(obj)
+		} else {
+			jData, err = json.Marshal(obj)
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/utils/utils.go:234]: We should keep the same way to report error. Why we omit the error in other places but keep it in this place?

Comment (lirenjie95) [reply api/pkg/apis/v1alpha1/utils/utils.go:234]: In the original version, we omit all errors. Should we keep it or record as many error as possible?

Comment (msftcoderdjw) [reply api/pkg/apis/v1alpha1/utils/utils.go:234]: I prefer we always return error except we think the caller doesn't need to know it.

**Review by msftcoderdjw — APPROVED** [review]

**Pull request merged**